src/ngraph/frontend/onnx_import/op/non_max_suppression.cpp

```diff
@@ -0,0 +1,89 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <memory>
+
+#include "ngraph/frontend/onnx_import/exceptions.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/non_max_suppression.hpp"
+#include "ngraph/op/util/attr_types.hpp"
+#include "non_max_suppression.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace set_1
+            {
+                NodeVector non_max_suppression(const Node& node)
+                {
+                    // TODO: this op will not be tested until at least
+                    //       a reference implementation is added
+
+                    const auto ng_inputs = node.get_ng_inputs();
+                    const std::shared_ptr<ngraph::Node> boxes = ng_inputs.at(0);
+                    const std::shared_ptr<ngraph::Node> scores = ng_inputs.at(1);
+
+                    std::shared_ptr<ngraph::Node> max_output_boxes_per_class =
```

Comment:
We don't need to create these constants by default.
You can create them in the `else` clause.